File: 455_AssignCookies.cpp
//94.63%
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        int res(0);
		sort(g.begin(), g.end());
		sort(s.begin(), s.end());
		int i(0), size(s.size());
		for(auto greed : g){
			while(i < size){
				if(s[i++] >= greed){
					res++;
					break;
				}
			}
			if(i >= size)
				break;
		}
		return res;
    }
};

File: 455_AssignCookies.py
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        res, i, size = 0, 0, len(s)
        g.sort()
        s.sort()
        for greed in g:
            while i < size:
                if s[i] >= greed:
                    res += 1
                    i += 1
                    break
                i += 1
            if i >= size:
                break
        return res               